ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/AbstractPartitionPlan.java

```diff
@@ -341,20 +341,18 @@ protected class BinPackingTaskSplitter implements TaskSplitter {
     @Override
     public List<SplitTask> splitTasks(int targetTaskCount) {
       // bin-packing
-      List<FileTask> allDataFiles = Lists.newArrayList();
-      segmentFiles.forEach((dataFile, deleteFiles) ->
-          allDataFiles.add(new FileTask(dataFile, deleteFiles, false)));
+      List<FileTask> allDataFiles = Lists.newLinkedList();
       fragmentFiles.forEach((dataFile, deleteFiles) ->
           allDataFiles.add(new FileTask(dataFile, deleteFiles, true)));
+      segmentFiles.forEach((dataFile, deleteFiles) ->
+          allDataFiles.add(new FileTask(dataFile, deleteFiles, false)));
 
-      long taskSize = config.getTargetSize();
-      Long sum = allDataFiles.stream().map(f -> f.getFile().fileSizeInBytes()).reduce(0L, Long::sum);
-      int taskCnt = (int) (sum / taskSize) + 1;
-      List<List<FileTask>> packed = new BinPacking.ListPacker<FileTask>(taskSize, taskCnt, true)
+      List<List<FileTask>> packed = new BinPacking.ListPacker<FileTask>(
+          config.getTargetSize(), Integer.MAX_VALUE, false)
```

Comment:
Why lookback set to Max

Reply:
Setting it to the maximum value ensures that the maximum number of bin sizes can be packed to match the target size.

Reply:
I think in most of cases, there are no difference between using `Integer.MAX_VALUE` or `taskCnt`.WDYT

Reply:
I am worried that Integer.MAX_VALUE may cause performance issues with binpacking when there are a large number of files. We can test it and if there are no problems, I think it's okay.

Reply:
Okay, do you think setting the lookback to `1` is acceptable?

Reply:
Okay, let's keep the original lookback size

Reply:
I agree with using `Integer.MAX_VALUE`, because `Integer.MAX_VALUE` and `taskCnt` are essentially the same, and `taskCnt` worked well before.